### ISSUE TITLE
Two Dimensional Histograms

### ISSUE THREAD

**Issue by demarley** [scikit-hep/uproot#92]
When I open a 2D histogram in uproot and convert the data to a numpy array, the output only contains bin edges for one axis and under/overflow bins are included in the content, e.g.,:

```
# h2 is the 2d histogram from a root file with 4 x-bins and 2 y-bins
>>> h2.numpy
(array([0.  , 0.  , 0.  , 0.  , 0.  , 0.  , 0.08, 0.06, 0.03, 0.02, 0.  ,
       0.  , 0.01, 0.04, 0.07, 0.1 , 0.  , 0.  , 0.  , 0.  , 0.  , 0.  ]), array([0., 1., 2., 3., 4.]))
```

The bins with 0 are somewhat consistent with the under/overflow bins (in ROOT the 'real' bins start at bin numbers 7-10; 13-16).  However, from the information returned by the `numpy` method, it isn't clear (without already knowing) where the 'real' bins begin and end.

Could the [TH1Methods](https://github.com/scikit-hep/uproot/blob/master/uproot/hist.py) class be extended to support 2D histograms such that the above output would instead be something like:
```
>>> h2.numpy
(
array( [0.08, 0.06, 0.03, 0.02],    # first y-bin
       [0.01, 0.04, 0.07, 0.1] ] ), # second y-bin
array( [ [0., 1., 2., 3., 4.], [0., 1., 2.] ]) . # (x-bins, y-bins) 
)
```

NB: I believe I can recreate what I want using the `allvalues`, `fXaxis`, `fYaxis`, `f*axis.fXmin`, `f*axis.fXmax`, and `f*axis.fNbins` methods. If the bins have variable widths, it appears the `f*axis.fXbins` method will contain the proper bin edges.

**Comment by jpivarski** [comment]
This sounds good. As you can see, I haven't explicitly thought about methods for 2d histograms. If you have in mind what kinds of methods you'd like to see, I've created a stub for TH2Methods.

If you check out (or fork) the [issue-92](https://github.com/scikit-hep/uproot/tree/issue-92) branch, you can add whatever you want in [this class](https://github.com/scikit-hep/uproot/blob/issue-92/uproot/hist.py#L264). Any methods or properties that depend only on the `f*` variables can be added to 2D histograms.

Here's a demo. The tests/samples/hepdata-example.root file contains a 2D histogram named "hpxpy" and a 1D histogram named "hpx":

```python
>>> import uproot      # local branch
>>> f = uproot.open("tests/samples/hepdata-example.root")
>>> f["hpxpy"].hello()
'world'
>>> f["hpx"].hello()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'TH1F' object has no attribute 'hello'
```

You can probably get the non-underflow/non-overflow bins with a `.reshape` (to make it 2-d) followed by a slice.

**Comment by jpivarski** [comment]
I merged your implementation into scikit-hep/uproot/issue-92 and made my own clean-ups and fixes. There's a pull request to add it to master with a new version number (2.8.33 → 2.9.0 because it's a new feature). I've given you commit access so that you can make changes directly on the pull request to put in any last-minute additions. For example, I had to add a `hist2d` function so that `TH2Methods.__add__` would be able to make new histograms, but I only put in enough functionality for `__add__` to work. You might want to make `hist2d` as fully functional as `hist`.

When you're satisfied, give me the go-ahead and I'll merge to master and post uproot 2.9.0 on PyPI. Thanks!

**Comment by jpivarski** [comment]
Pull request #94, that is.

**Comment by demarley** [comment]
Closed by #94

**Comment by jpivarski** [comment]
Version 2.9.0 [is on PyPI](https://pypi.org/project/uproot/). Cheers!
